File: examples/hello/hello.rb
require 'json'

name = "World"

payload = ENV['PAYLOAD']
if payload == nil || payload == "" 
    payload = STDIN.read
end

STDERR.puts 'payload: ' + payload.inspect
if payload != ""
    payload = JSON.parse(payload)
    name = payload['name']
end

puts "Hello #{name}!"

File: images/hello/hello.go
package main

import (
	"encoding/json"
	"fmt"
	"os"
)

type person struct {
	Name string
}

func main() {
	n := os.Getenv("NAME") // can grab name from env or input
	if n == "" {
		n = "World"
	}
	p := &person{Name: n}
	json.NewDecoder(os.Stdin).Decode(p)
	fmt.Printf("Hello %v!\n", p.Name)
}
